src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -318,6 +318,32 @@ internal bool IsEnoughMemoryToScale(long hostProcessSize, IEnumerable<long> work
             return true;
         }
 
+        /// <summary>
+        /// Checks if a new worker can be added.
+        /// </summary>
+        /// <param name="workerChannels">All current worker chanells.</param>
+        /// <returns>True if a new worker can be started.</returns>
+        internal bool CanScale(IEnumerable<IRpcWorkerChannel> workerChannels)
+        {
+            // Cancel if there is any "non-ready" channel.
+            var nonreadyWorkerChannels = workerChannels.Where(x => x.IsChannelReadyForInvocations() == false);
+            if (nonreadyWorkerChannels.Any())
+            {
+                _logger.LogDebug($"Starting new language worker canceled: TotalChannels={workerChannels.Count()}, NonReadyChannels={nonreadyWorkerChannels.Count()}");
```

Comment:
We should be using structured logging for all _logger statements

```suggestion
                _logger.LogDebug("Starting new language worker canceled: TotalChannels={workerChannelCount}, NonReadyChannels={nonReadyChannelCount}", workerChannels.Count(), nonreadyWorkerChannels.Count());
```

Reply:
Also, if it makes sense to add the reason for cancellation in log messages. This will be helpful when debugging through kusto logs. For example -
`Starting new language worker canceled as there is atleast one non ready channel`  OR
`Starting new language worker canceled as the count of total channels reaches the maximum limit`